bundles/org.openhab.core.automation/src/main/java/org/openhab/core/automation/internal/module/handler/ItemStateUpdateActionHandler.java

```diff
@@ -0,0 +1,99 @@
+/**
+ * Copyright (c) 2010-2020 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.core.automation.internal.module.handler;
+
+import java.util.Map;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.core.automation.Action;
+import org.openhab.core.automation.handler.BaseActionModuleHandler;
+import org.openhab.core.config.core.Configuration;
+import org.openhab.core.events.EventPublisher;
+import org.openhab.core.items.Item;
+import org.openhab.core.items.ItemNotFoundException;
+import org.openhab.core.items.ItemRegistry;
+import org.openhab.core.items.events.ItemEventFactory;
+import org.openhab.core.items.events.ItemStateEvent;
+import org.openhab.core.types.State;
+import org.openhab.core.types.TypeParser;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * This is an implementation of an ActionHandler. It sends state updates for items.
+ *
+ * @author Christoph Weitkamp - Initial contribution
+ */
+@NonNullByDefault
+public class ItemStateUpdateActionHandler extends BaseActionModuleHandler {
+
+    public static final String ITEM_STATE_UPDATE_ACTION = "core.ItemStateUpdateAction";
+    public static final String ITEM_NAME = "itemName";
+    public static final String STATE = "state";
+
+    private final Logger logger = LoggerFactory.getLogger(ItemStateUpdateActionHandler.class);
+
+    private final EventPublisher eventPublisher;
+    private final ItemRegistry itemRegistry;
+
+    private final String itemName;
+    private final String state;
+
+    public ItemStateUpdateActionHandler(Action module, EventPublisher eventPublisher, ItemRegistry itemRegistry) {
+        super(module);
+
+        this.eventPublisher = eventPublisher;
+        this.itemRegistry = itemRegistry;
+
+        final Configuration config = module.getConfiguration();
```

Comment:
Looking closer into the code, I am not 100% certain that the module configuration cannot be changed during the lifetime of the handler. I at least found `ConfigurationNormalizer.normalizeConfiguration`, which reads the configuration object and modifies it. I'd hence feel safer if this would be re-read in `execute`, just as it is done for the `ItemCommandActionHandler`.

Reply:
I applied these suggestions. But I have to investigate later on that because IIRC I contributed other automation modules doing it in the same way.